docs/data-sources/hss_hosts.md

```diff
@@ -0,0 +1,147 @@
+---
+subcategory: "Host Security Service (HSS)"
+---
+
+# huaweicloud_hss_hosts
+
+Use this data source to get the list of HSS hosts within HuaweiCloud.
+
+## Example Usage
+
+```hcl
+variable host_id {}
+
+data "hss_hosts" "test" {
+  host_id = var.host_id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the HSS hosts.
+  If omitted, the provider-level region will be used.
+
+* `host_id` - (Optional, String) Specifies the ID of the host to be queried.
+
+* `name` - (Optional, String) Specifies the name of the host to be queried.
+  This field will undergo a fuzzy matching query, the query result is for all hosts whose names contain this value.
+
+* `status` - (Optional, String) Specifies the status of the host to be queried.  
+  The valid values are as follows:
+  + **ACTIVE**
+  + **SHUTOFF**
+  + **BUILDING**
+  + **ERROR**
+
+* `os_type` - (Optional, String) Specifies the operating system type of the host to be queried.  
+  The valid values are as follows:
+  + **Linux**
+  + **Windows**
+
+* `agent_status` - (Optional, String) Specifies the agent status of the host to be queried.  
+  The valid values are as follows:
+  + **installed**
+  + **not_installed**
+  + **online**
+  + **offline**
+  + **install_failed**
+  + **installing**
+
+* `protect_status` - (Optional, String) Specifies the protection status of the host to be queried.  
+  The valid values are as follows:
+  + **closed**
+  + **opened**
+
+* `protect_version` - (Optional, String) Specifies the protection version enabled by the host to be queried.  
+  The valid values are as follows:
+  + **hss.version.basic**
+  + **hss.version.advanced**
+  + **hss.version.enterprise**
+  + **hss.version.premium**
+  + **hss.version.wtp**
+  + **hss.version.container.enterprise**
+
+* `protect_charging_mode` - (Optional, String) Specifies the charging mode for the host protection quota to be queried.
+  
+  The valid values are as follows:
+  + **prePaid**
+  + **postPaid**
+
+* `detect_result` - (Optional, String) Specifies the security detection result of the host to be queried.  
+  The valid values are as follows:
+  + **undetected**
+  + **clean**
+  + **risk**
+  + **scanning**
+
+* `group_id` - (Optional, String) Specifies the host group ID to which the hosts to query belong.
+
+* `policy_group_id` - (Optional, String) Specifies the policy group ID to which the hosts to query belong.
+
+* `asset_value` - (Optional, String) Specifies the asset importance of the host to be queried.  
+  The valid values are as follows:
+  + **important**
+  + **common**
+  + **test**
+
+* `enterprise_project_id` - (Optional, String) Specifies the ID of the enterprise project to which the hosts belong.
+  If omitted, will query the values under the default enterprise project. Setting it to **all_granted_eps** will query
+  the hosts under all enterprise projects.
```

Comment:
seems the default value should be **all_granted_eps** in the function logic

Reply:
done.